demos/provider-mock/src/api.ts

```diff
@@ -85,5 +88,153 @@ export function prosopoRouter(): Router {
 		},
 	);
 
+	router.get("/test", async (req, res) => {
+		console.log(req.headers);
+		console.log(req.rawHeaders);
+
+		// Extract headers
+		const headers = req.headers;
+
+		// Validate headers
+		if (
+			!headers["x-tls-clienthello"] ||
+			!headers["x-tls-version"] ||
+			!headers["x-tls-server-name"]
+		) {
+			console.error("Missing required headers.");
+			return res.status(400).send("Missing required headers.");
+		}
+
+		// make sure the client hello message is a string
+		if (
```

Comment:
aren't headers always string-string key/value maps?

Reply:
set to default empty strings if not present